File: cp/BinaryEquivalent.cpp
#include<bits/stdc++.h>
#define num long int
using namespace std;

int main()
{
	//ios_base::sync_with_stdio(0);
	//cin.tie(0); cout.tie(0);

	short n; cin>>n;

	num arr[n],ones[n]={0},max=0,sum=0,c=0;

	for(num i=0; i<n; i++)
	{
		cin>>arr[i]; if(arr[i]>max) max = arr[i];

		bitset<100000> b(arr[i]);

		ones[i] = b.count();
	}

	//for(num i=0; i<n; i++)
	//	cout<<ones[i]<<" ";

	max = int(log2(max)) + 1;

	//1
	for(num i=0; i<n; i++)
		if(ones[i]*2 == max)	c++;

	//2
	for(num i=0; i<n; i++)
	{
		
	}

	return 0;
}

File: cp/BinaryEquivalent.py
from itertools import combinations
from math import log2 
from sys import stdout,stdin

n = int(stdin.readline())
arr = list(map(int,stdin.readline().split( )))

ones,m = [],0

for i in range(n):

	ones.append(bin(arr[i]).count('1'))

	if(arr[i]>m):
		m = arr[i]

m = int(log2(m)+1)

#print(m)
count = 0

for i in range(n):

	if(ones[i]*2 == m):

		count+=1

for i in range(2,n+1):

	arr = list(combinations(ones,i))

	for j in arr:

		if(sum(j)*2 == m*i):

			count+=1

ans = bin(count).replace('0b','')

m = m-len(ans)

tmp = ''
for i in range(m):
	tmp+='0'
tmp += ans

stdout.write(tmp+"\n")